src/cc1101.cpp

```diff
@@ -443,29 +524,174 @@ void cc1101::CCinit(void) {                                   // initialize CC11
+	DBG_PRINT(F("POR done,"));
 	delay(10);
 
 	cc1101_Select();
 	DBG_PRINT(FPSTR(TXT_EEPROM)); 	DBG_PRINT(FPSTR(TXT_BLANK));	DBG_PRINT(FPSTR(TXT_READ));
-	wait_Miso();                                              // Wait until MISO goes low
+	wait_Miso();                                           // Wait until MISO goes low
 
 	sendSPI(0x00 | CC1101_WRITE_BURST);
-	for (uint8_t i = 0; i < sizeof(cc1101::initVal); i++) {   // write EEPROM value to cc1101
+	for (uint8_t i = 0; i < sizeof(cc1101::initVal); i++) {         // write EEPROM value to cc1101
 		sendSPI(EEPROM.read(EE_CC1101_CFG + i));
-		DBG_PRINT(".");
-		
+		DBG_PRINT('.');
 	}
+
 	cc1101_Deselect();
-	delayMicroseconds(10);            // ### todo: welcher Wert ist als delay sinnvoll? ###
+	delayMicroseconds(10);                                          // ### todo: welcher Wert ist als delay sinnvoll? ###
+
+	ccmode = ( (EEPROM.read(EE_CC1101_CFG + 18) ) & 0x70 ) >> 4;    // first read modulation direct from 0x12 MDMCFG2
 
-	writePatable();                   // write PatableArray to patable reg
+	writePatable();                                                 // write PatableArray to patable reg
 	DBG_PRINTLN(F("done"));
 
 	delay(1);
 	setReceiveMode();
 
 #endif
 }
+
+
+void cc1101::getRxFifo(uint16_t Boffs) {           // xFSK
+	uint8_t fifoBytes;
+	bool dup;                                      // true bei identischen Wiederholungen bei readRXFIFO
+
+	if (isHigh(PIN_RECEIVE)) {                     // wait for CC1100_FIFOTHR given bytes to arrive in FIFO
+		#ifdef PIN_LED_INVERSE
+			digitalWrite(PIN_LED, !blinkLED);
+		#else
+			digitalWrite(PIN_LED, blinkLED);
+		#endif
+
+/*
+ * 
+ * Ralf ( mode numbering == own selection )
+ * cc1101 Mode: 0 - normal ASK/OOK, 1 - FIFO, 2 - FIFO ohne dup, 3 - FIFO LaCrosse, 4 - experimentell, 9 - FIFO mit Debug Ausgaben
+ *
+ * Sidey ( mode numbering == cc1101 data sheet setting numbering 0x12: MDMCFG2–Modem Configuration )
+ * cc1101 Mode: 0 - FIFO LaCrosse, 3 - normal ASK/OOK
+ * 
+
+    if (ccmode == 4) {
+      cc1101::ccStrobe_SIDLE(); // start over syncing
+    }
+*/
+
+		fifoBytes = cc1101::getRXBYTES();          // & 0x7f; // read len, transfer RX fifo
+		if (fifoBytes > 0) {
+			uint8_t marcstate;
+			uint8_t RSSI = cc1101::getRSSI();
+
+/*
+ * !!! for DEVELOPMENT and DEBUG only !!!
+ * 
+      #ifdef DEBUG
+        if (cc1101::ccmode == 0) {
+          MSG_PRINT(F("RX fifoBytes ("));
+          MSG_PRINT(fifoBytes);
+          MSG_PRINTLN((") "));
+        }
+      #endif
+ * 
+ */
+
+			if (fifoBytes < 0x80) {                // RXoverflow?
+				if (fifoBytes > ccMaxBuf) {
+					fifoBytes = ccMaxBuf;
+				}
+				dup = readRXFIFO(fifoBytes);
+				if (cc1101::ccmode != 2 || dup == false) {
+
+					if (cc1101::ccmode != 9) {
+						MSG_PRINT(char(MSG_START));      // SDC_WRITE not work in this scope
+						MSG_PRINT(F("MN;D="));
+					}
+					for (uint8_t i = 0; i < fifoBytes; i++) {
+            MSG_PRINTtoHEX(ccBuf[i]);
```

Comment:
hier passt was mit dem Einrücken nicht.

Reply:
Gern passe ich die Einrückung an aber wenn du genau HIER darauf Wert legst, so möchte ich GERN alle Einrückungen auch mal überarbeiten ;-)

Reply:
Die Einrückung expliziet hier, habe ich angepasst.

Ich würde gern, nachdem dieser PR übernommen wurde, eine komplette Überarbeitung der Einrückungen "durchgehen" weil das in einigen Dateien nicht übereinstimmt.